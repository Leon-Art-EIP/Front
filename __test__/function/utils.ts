import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
