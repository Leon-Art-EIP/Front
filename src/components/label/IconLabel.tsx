import { SvgIconProps } from "@mui/material";

export interface IIconLabelProps {
  icon: React.ComponentType<SvgIconProps>;
  text: string;
}

export default function IconLabel({ icon: Icon, ...props }: IIconLabelProps): JSX.Element {
  return (
    <div className="flex gap-8 items-center align-middle h-12">
      <Icon style={{ fontSize: 48 }} />
      <span className="font-medium text-xl">{props.text}</span>
    </div>
  );
}
