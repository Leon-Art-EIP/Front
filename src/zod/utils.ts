export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
