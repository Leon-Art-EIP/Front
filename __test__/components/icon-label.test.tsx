import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import IconLabel from "../../src/components/label/IconLabel";
import { Upcoming } from "@mui/icons-material";

test("IconLabel snapshot", () => {
  testSnapshot(<IconLabel icon={Upcoming} text="test" />);
});
