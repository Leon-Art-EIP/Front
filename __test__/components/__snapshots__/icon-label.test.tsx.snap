// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconLabel snapshot 1`] = `
<div
  className="flex gap-8 items-center align-middle h-12"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-inherit css-i4bv87-MuiSvgIcon-root"
    data-testid="UpcomingIcon"
    focusable="false"
    style={
      {
        "fontSize": 48,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="m21.16 7.26-1.41-1.41-3.56 3.55 1.41 1.41s3.45-3.52 3.56-3.55zM11 3h2v5h-2zm-4.6 7.81L7.81 9.4 4.26 5.84 2.84 7.26c.11.03 3.56 3.55 3.56 3.55zM20 12h-5c0 1.66-1.34 3-3 3s-3-1.34-3-3H4c-1.1 0-2 .9-2 2v5c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-5c0-1.1-.9-2-2-2z"
    />
  </svg>
  <span
    className="font-medium text-xl text-inherit"
  >
    test
  </span>
</div>
`;
